### PR TITLE
Add index page on HTTP Server

### PR DESCRIPTION
**Pull request by AndrewDi**
### What problem does this PR solve? 
Add index page on HTTP Server, a little more convenient to user.

### What is changed and how it works?

Tests 
 - Manual test (add detailed scripts or steps below)
http://localhost:10080 display generate index page

Code changes

Side effects
none

Related changes

### CODE REVIEW

**Comment by shenli** [comment]
@AndrewDi Thanks!

File: server/http_status.go [hunk 1]
```diff
@@ -123,3 +125,24 @@ func (s *Server) handleStatus(w http.ResponseWriter, req *http.Request) {
 		terror.Log(errors.Trace(err))
 	}
 }
+
+func (s *Server) handleHTTPRoot(w http.ResponseWriter, req *http.Request) {
```
Comment (shenli) [inline server/http_status.go]: It is better to use the [Router.Walks()](https://github.com/gorilla/mux#walking-routes) than hardcode.

Comment (AndrewDi) [reply server/http_status.go]: if I use Router.Walks(), I can not get /debug/pprof/ and can not get a more nice name for router, such as BinlogRecover for /binlog/recover.

Comment (shenli) [reply server/http_status.go]: The second one is reasonable.

File: server/http_status.go [hunk 2]
```diff
@@ -123,3 +125,24 @@ func (s *Server) handleStatus(w http.ResponseWriter, req *http.Request) {
...
+<tr><td><a href='/binlog/recover'>BinlogRecover</a><td></tr>
+<tr><td><a href='/schema'>Schema</a><td></tr>
+<tr><td><a href='/info'>Info</a><td></tr>
+<tr><td><a href='/info/all'>InfoAll</a><td></tr>
+<tr><td><a href='/debug/pprof/'>Debug</a><td></tr>
+</table>
+</body>
+</html>
+`)
+	w.Write(landingPage)
```
Comment (lamxTyler) [inline server/http_status.go]: Do we need to handle the error here?

**Comment by AndrewDi** [comment]
I remove hardcode, and add name attribute to Routers.

File: server/http_status.go [hunk 3]
```diff
@@ -27,6 +25,8 @@ import (
 	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus"
 	log "github.com/sirupsen/logrus"
+	"net/http"
+	"net/http/pprof"
```
Comment (jackysp) [inline server/http_status.go]: Please move the system libraries to its own part.

**Comment by jackysp** [comment]
LGTM
PTAL @coocood

File: server/http_status.go [hunk 4]
```diff
@@ -85,9 +87,38 @@ func (s *Server) startHTTPServer() {
...
+	})
+	if err != nil {
+		fmt.Println(err)
+	}
+	httpRouterPage.WriteString("<tr><td><a href='/debug/pprof/'>Debug</a><td></tr>")
+	httpRouterPage.WriteString("</table></body></html>")
+	router.HandleFunc("/", func(responseWriter http.ResponseWriter, request *http.Request) {
+		_, err = responseWriter.Write([]byte(httpRouterPage.String()))
+		if err != nil {
+			log.Error("Http index page error", err)
```
Comment (zimulala) [inline server/http_status.go]: Do we need to add a blank space after "page error"?

File: server/http_status.go [hunk 5]
```diff
@@ -85,9 +87,38 @@ func (s *Server) startHTTPServer() {
...
+		httpRouterPage bytes.Buffer
+		pathTemplate   string
+	)
+	httpRouterPage.WriteString("<html><head><title>TiDB Status and Metrics Report</title></head><body><h1>TiDB Status and Metrics Report</h1><table>")
+	err = router.Walk(func(route *mux.Route, router *mux.Router, ancestors []*mux.Route) error {
+		pathTemplate, err = route.GetPathTemplate()
+		if err != nil {
+			log.Error("Get http router path err", err)
+		}
+		name := route.GetName() //if not set name attribute,GetName return ""
```
Comment (zimulala) [inline server/http_status.go]: s/`if not set name attribute,GetName return ""`/`If the name attribute is not set, GetName returns "".`

File: server/http_status.go [hunk 6]
```diff
@@ -85,9 +87,38 @@ func (s *Server) startHTTPServer() {
...
+		if name != "" && err == nil {
+			httpRouterPage.WriteString("<tr><td><a href='" + pathTemplate + "'>" + name + "</a><td></tr>")
+		}
+		return nil
+	})
+	if err != nil {
+		fmt.Println(err)
```
Comment (zimulala) [inline server/http_status.go]: I think “log” is better to use.

Comment (AndrewDi) [reply server/http_status.go]: OMG,I forgot it...

File: server/http_status.go (hunk 4 above)
Comment (AndrewDi) [reply server/http_status.go]: OK

**Comment by AndrewDi** [comment]
PTAL @coocood @zimulala

**Comment by jackysp** [comment]
PTAL @zimulala

**Pull request merged**